### PR TITLE
wasm-benchmarks: apply lateral joins when possible, and return rel fields in results 

### PR DESCRIPTION
**Pull request by miguelff**
Needed for https://github.com/prisma/team-orm/issues/910

### CODE REVIEW

**Comment by miguelff** [comment]
This is benchmarks-only code. I'm piggybacking on https://github.com/prisma/prisma-engines/pull/4689

File: query-engine/driver-adapters/executor/src/bench.ts [hunk 1]
```diff
@@ -57,23 +64,36 @@ async function recordQueries(
   datamodel: string,
   prismaQueries: any
 ): Promise<void> {
-  const qe = await initQeWasmBaseLine(adapter, datamodel);
-  await qe.connect("");
+  // Different engines might have made different SQL queries to complete the same Prisma Query,
+  // so we record the results of all engines and assert that they are the same.
```
Comment (jkomyno) [inline query-engine/driver-adapters/executor/src/bench.ts:68]: Where are we asserting that the result of each Query Engine is the same?

Comment (miguelff) [reply query-engine/driver-adapters/executor/src/bench.ts:68]: Leftover, removed this as it increased notably CPU time and skewed results

File: query-engine/driver-adapters/executor/src/bench.ts [hunk 2]
```diff
@@ -57,23 +64,37 @@ async function recordQueries(
...
+  // Different engines might have made different SQL queries to complete the same Prisma Query,
+  // so we record the results of all engines for the benchmarking phase.
+  const napi = await initQeNapiCurrent(adapter, datamodel);
+  await napi.connect("");
+  const wasmCurrent = await initQeWasmCurrent(adapter, datamodel);
+  await wasmCurrent.connect("");
+  const wasmBaseline = await initQeWasmBaseLine(adapter, datamodel);
+  await wasmBaseline.connect("");
+  const wasmLatest = await initQeWasmLatest(adapter, datamodel);
+  await wasmLatest.connect("");
```
Comment (jkomyno) [inline query-engine/driver-adapters/executor/src/bench.ts:76]: nit: Given that these `initQe*` have the same API (and all call the `.connect("")` method first and `.disconnect("")` method last), one could rationalise this code by mapping over the `[napi, wasmCurrent, wasmBaseline, wasmLatest]` before constructing the Query Engine instances.

However, this code could change in the future due to API differences between e.g. Napi and Wasm, so I'll approve this PR, leaving to @miguelff the judgement about whether to apply "DRY" or not in this case :)

**Review by jkomyno — APPROVED** [review]
Approved with a nit. Thanks!

**Pull request merged**